src/integration_tests/merge_subscription.rs

```diff
@@ -0,0 +1,223 @@
+                    name: update_mock.to_string(),
+                    merge_strategy: Some(MergeStrategy::MergeStorageChanges),
+                },
+            ],
+            aliases: vec![],
+        },
+    };
+
+    let (addr, server) = start_server(
+        &config,
+        Client::new(&[format!("ws://{addr}")]).await.unwrap(),
+    )
+    .await
+    .unwrap();
+
+    let client = Client::new(&[format!("ws://{addr}")]).await.unwrap();
+    let mut first_sub = client
+        .subscribe(subscribe_mock, vec![], unsubscribe_mock)
+        .await
+        .unwrap();
+
+    tokio::spawn(async move {
+        let (_, mock_sub_sink) = mock_sub_rx.recv().await.unwrap();
+
+        run_sink_tasks(
+            &mock_sub_sink,
+            vec![
+                SinkTask::Send(json!({
+                    "block": "0x01",
+                    "changes": [
+                        ["0x01", "hello"],
+                        ["0x02", null]
+                    ]
+                })),
+                SinkTask::Sleep(1_000),
+                SinkTask::Send(json!({
+                    "block": "0x02",
+                    "changes": [
+                        ["0x02", "world"]
+                    ]
+                })),
+                SinkTask::Sleep(1_000),
+                SinkTask::Send(json!({
+                    "block": "0x03",
+                    "changes": [
+                        ["0x01", null],
+                        ["0x02", "bye"]
+                    ]
+                })),
+                SinkTask::Sleep(1_000),
+                SinkTask::Send(json!({
+                    "block": "0x04",
+                    "changes": [
+                        ["0x01", "hello"],
+                        ["0x02", "again"]
+                    ]
+                })),
+            ],
+        )
+        .await;
+    });
+
+    let test_one = tokio::spawn(async move {
+        assert_eq!(
+            first_sub.next().await.unwrap().unwrap(),
+            json!({
+                "block": "0x01",
+                "changes": [
+                    ["0x01", "hello"],
+                    ["0x02", null]
+                ]
+            })
+        );
+
+        assert_eq!(
+            first_sub.next().await.unwrap().unwrap(),
+            json!({
+                "block": "0x02",
+                "changes": [
+                    ["0x02", "world"],
+                ]
+            })
+        );
+
+        assert_eq!(
+            first_sub.next().await.unwrap().unwrap(),
+            json!({
+                "block": "0x03",
+                "changes": [
+                    ["0x01", null],
+                    ["0x02", "bye"]
+                ]
+            })
+        );
+
+        // first subscription will unsubscribe but it shouldn't affect second subscription
+        first_sub.unsubscribe().await.unwrap();
+    });
+
+    tokio::time::sleep(std::time::Duration::from_millis(1_500)).await;
```

Comment:
why 1.5s wait? I want the tests ran as fast as possible so if we need to make it sleep, it should be minimal time

Reply:
I'm sending msg every seconds so 1.5s is just before 3rd msg send. Probably need to reduce time to millis